src/basic/io-util.c

```diff
@@ -107,6 +107,18 @@ int loop_read_exact(int fd, void *buf, size_t nbytes, bool do_poll) {
         return 0;
 }
 
+int loop_write_exact(int fd, void *buf, size_t nbytes) {
+        ssize_t n;
+
+        n = loop_write(fd, buf, nbytes);
+        if (n < 0)
+                return (int) n;
+        if ((size_t) n != nbytes)
+                return -EIO;
+
+        return 0;
+}
```

Comment:
no need to define this. loop_write() already works like this.

i.e. loop_read() and loop_write() are not symmetric regarding short reads/writes. loop_read() will treat them gracefully because EOF is a thing on read. But loop_write() considers any short writes as failures already.